MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -447,6 +453,416 @@ void EnggDiffractionPresenter::processRebinMultiperiod() {
+    RebinToWs->setProperty("WorkspaceToMatch", "engggui_fitting_focused_ws");
+    RebinToWs->setProperty("OutputWorkspace", workspaceName);
+    RebinToWs->execute();
+  } catch (std::runtime_error &re) {
+    g_log.error() << "Could not run the algorithm RebinToWorkspace, "
+                     "Error description: " +
+                         static_cast<std::string>(re.what())
+                  << std::endl;
+  }
+}
+
+void EnggDiffractionPresenter::plotFitPeaksCurves() {
+  AnalysisDataServiceImpl &ADS = Mantid::API::AnalysisDataService::Instance();
+  std::string singlePeaksWs = "engggui_fitting_single_peaks";
+  std::string focusedPeaksWs = "engggui_fitting_focused_ws";
+
+  if (ADS.doesExist(singlePeaksWs) && ADS.doesExist(focusedPeaksWs)) {
+    auto focusedPeaksWS = ADS.retrieveWS<MatrixWorkspace>(focusedPeaksWs);
+    auto singlePeaksWS = ADS.retrieveWS<MatrixWorkspace>(singlePeaksWs);
+    try {
+      auto focusedData = ALCHelper::curveDataFromWs(focusedPeaksWS);
+      m_view->setDataVector(focusedData, true);
+      auto singlePeaksData = ALCHelper::curveDataFromWs(singlePeaksWS);
+      m_view->setDataVector(singlePeaksData, false);
+
+    } catch (std::runtime_error &re) {
+      g_log.error() << "Unable to finish of the plotting of the graph for "
+                       "engggui_fitting_focused_fitpeaks  workspace. Error "
+                       "description : " +
+                           static_cast<std::string>(re.what()) +
+                           " Please check also the log message for detail.";
+      throw;
+    }
+  } else {
+    g_log.error() << "Fitting could not be plotted as there is no " +
+                         singlePeaksWs + " or " + focusedPeaksWs +
+                         " workspace found."
+                  << std::endl;
+  }
+}
+
+void EnggDiffractionPresenter::fittingFinished() {
+  if (!m_view)
+    return;
+
+  if (!m_fittingFinishedOK) {
+    g_log.warning() << "The single peak fitting did not finish correctly."
+                    << std::endl;
+    if (m_workerThread) {
+      delete m_workerThread;
+      m_workerThread = NULL;
+    }
+
+  } else {
+    g_log.notice() << "The single peak fitting finished - the output "
+                      "workspace is ready."
+                   << std::endl;
+
+    if (m_workerThread) {
+      delete m_workerThread;
+      m_workerThread = NULL;
+    }
+
+    g_log.notice()
+        << "EnggDiffraction GUI: plotting peaks for single peak fits "
+           "has started... "
+        << std::endl;
+    try {
+      plotFitPeaksCurves();
+
+    } catch (std::runtime_error &re) {
+      g_log.error() << "Unable to finish of the plotting of the graph for "
+                       "engggui_fitting_focused_fitpeaks workspace. Error "
+                       "description : " +
+                           static_cast<std::string>(re.what()) +
+                           " Please check also the log message for detail.";
+      throw;
```

Comment:
I think this re-throw is dangerous. It will probably crash if for some strange reason the plotting fails.
I'd say that giving an error message is enough, and all you can do here.